### PR TITLE
fix: linter warnings for connected_components_with_dsu.

### PR DESCRIPTION
**Pull request by fhlasek**
#### Description of Change
Splitting of from #954 (@ayaankhan98)

#### Checklist

- [X] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [X] Search previous suggestions before making a new one, as yours may be a duplicate.
- [X] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/992"></a>

### CODE REVIEW

File: graph/connected_components_with_dsu.cpp [hunk 1]
```diff
@@ -2,50 +2,52 @@
 #include <set>
 #include <vector>
 
-int N;  // denotes number of nodes;
+int number_of_nodes;  // denotes number of nodes;
 std::vector<int> parent;
-std::vector<int> siz;
+std::vector<int> connected_set_size;
```
Comment (Panquesito7) [inline graph/connected_components_with_dsu.cpp:7]: Avoid using global variables/macros.
Pass them as local function parameters.

File: graph/connected_components_with_dsu.cpp [hunk 2]
```diff
@@ -2,50 +2,52 @@
...
         parent[b] = a;
-        siz[a] += siz[b];
+        connected_set_size[a] += connected_set_size[b];
     }
 }
 
 int no_of_connected_components() {  // To find total no of connected components
     std::set<int> temp;  // temp set to count number of connected components
-    for (int i = 1; i <= N; i++) temp.insert(find_set(i));
+    for (int i = 1; i <= number_of_nodes; i++) temp.insert(find_set(i));
```
Comment (Panquesito7) [inline graph/connected_components_with_dsu.cpp:36]:
```suggestion
    for (int i = 1; i <= number_of_nodes; i++) { 
        temp.insert(find_set(i));
    }
```

**Comment by kvedala** [comment]
@fhlasek may I suggest to enable "GitHub Actions" on your fork of this repo. That way, much of the code formatting will be taken care of by the CI tools and make contributing more convenient for you.

**Comment by fhlasek** [comment]
@kvedala thanks for the suggestion, I'm going to enable it!

**Pull request merged**